File: articles/decision-science-and-democratic-public-reasoning/r/run_all_democratic_reasoning_workflows.R
# run_all_democratic_reasoning_workflows.R
# Run all R workflows for Decision Science and Democratic Public Reasoning.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "democratic_public_reasoning_workflow.R",
  "democratic_process_profiles.R",
  "public_trust_review.R",
  "contestability_review_tables.R",
  "democratic_public_reasoning_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R democratic-public-reasoning workflows completed.")

File: articles/decision-science-and-democratic-public-reasoning/python/run_all_democratic_reasoning_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Decision Science and Democratic Public Reasoning."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "legitimacy_model.py",
    "contestability_model.py",
    "public_trust_model.py",
    "democratic_process_comparison.py",
    "decision_record_exporter.py",
    "democratic_public_reasoning_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python democratic-public-reasoning workflows completed.")


if __name__ == "__main__":
    main()
